### ISSUE TITLE
Expand "Operation" model to allow stateful (numerical) actions on Families?

### ISSUE THREAD

**Issue by mdworken** [rubyforgood/abalone#146]
Details/Discussion TK. 
tl;dr: I think we want to explicitly avoid storing an "animal_count" on families because it will lead to really annoying issues syncing between Operations and Measurements, and we'd like to keep Measurements fully NOT stateful.

**Example Problem that can arise**

1. I count 500 animals and store that info on the Family model statefully.
2. I remove 10 animals for disease testing. The count should now clearly be 490.
3. I count 400 animals the next week. The count is 400.
4. I remove 10 more animals for disease testing. The count is 390.
5. BUT THEN I realize I had a data entry error, and there were actually 600 animals at the start, not 500.

Q: How many animals should I have when I "replay" these events with the change from 500->600 How does that answer change if the ordering between 3/4 flips, or is ambiguous?

Potential Answer: That's a huge headache to figure out and maintain the logic for - let's try and avoid the whole situation if feasible.

**Acceptance Criterion**
- [ ] Group comes to a conclusion on this issue.
- [ ] Summary of the discussion is posted to this ticket.
- [ ] Architectural Decision Record is created and put somewhere accessible, and that ADR includes this ticket.
